_posts/en/newsletters/2025-01-31-newsletter.md

```diff
@@ -0,0 +1,165 @@
+  htlc], LDK will attempt to resolve as many HTLCs as possible in the
+  same transaction to save on transaction fees.  However, if the channel
+  counterparty is able confirm any of the batched HTLCs first, that will
+  _conflict_ with the batch transaction and make it invalid.  In that
+  case, LDK would correctly create an updated batch transaction with the
+  conflict removed.  Unfortunately, if the counterparty's transaction
+  conflicts with multiple separate batches, LDK would incorrectly only
+  update the first batch.  The remaining batches would not be able to
+  confirm.
+
+  Some HTLCs need to be resolved before a deadline or the counterparty
+  can steal back their funds.  The counterparty is prevented from
+  spending those HTLCs before the deadline by a [timelock][topic
+  timelocks].  Most older versions of LDK put those HTLCs in a separate
+  batch that it was sure to confirm before the counterparty could
+  confirm a conflicting transaction, ensuring no funds could be stolen.
+  For HTLCs that didn't allow funds theft, but which the counterparty
+  could resolve immediately, there was a risk that the counterparty
+  could cause funds to become stuck.  Morehouse writes that this can be
+  fixed by "upgrading to LDK version 0.1 and replaying the sequence of
+  commitment and HTLC transactions that led to the lock up."
+
+  However, the release candidate LDK 0.1-beta changed its logic (see
+  [Newsletter #335][news335 ldk3340]) and began batching all types of
+  HTLCs together, which could allow an attacker to create a conflict
+  with a timelocked HTLC.  If resolution of that HTLC remained stuck
+  after the timelock expired, theft was possible.  Upgrading to the
+  release version of LDK 0.1 also fixes this form of the vulnerability.
+
+  Morehouse's post provides additional details and discusses possible
+  ways to prevent future vulnerabilities stemming from the same root
+  cause.
+
+- **Replacement cycling attacks against miners:** Antoine Riard
+  [posted][riard minecycle] to the Bitcoin-Dev mailing list to
+  disclosure an additional vulnerability possible with the [replacement
+  cycling][topic replacement cycling] attack he originally publicly
+  disclosed in 2023 (see [Newsletter #274][news274 cycle]).  In short:
+
+  1. Bob broadcasts a transaction paying Mallory (and possibly other
+     people).
+
+  2. Mallory [pins][topic transaction pinning] Bob's transaction.
+
+  3. Bob doesn't realize he's been pinned and bumps the fees (using
+     either [RBF][topic rbf] or [CPFP][topic cpfp]).
+
+  4. Because Bob's original transaction was pinned, his fee bump
+     doesn't propagate.  However, Mallory receives it somehow.  Steps 3
+     and 4 may be repeated multiple times using replacement cycling to
+     greatly increase Bob's fee.
+
+  5. Mallory mines Bob's highest fee bump, which nobody else is trying
+     to mine because it didn't propagate.  This allows Mallory to earn
```

Comment:
```suggestion
     to mine because it didn't propagate, allowing her to earn
```

Reply:
Leaving as-is.  Prefer to limit sentences to one appositive phrase.